### PR TITLE
Tests of vcat of CategoricalArray and NullableCategoricalArray

### PR DESCRIPTION
**Pull request by gustafsson**
They are implemented together so at first I figured it would make sense to test them together. But then I realised that you've split all other tests of Nullable to its own file. The code is identical though.

The test of CategoricalArray can also be moved out of the nested for-loop in 11_array.jl since it doesn't use those parameters.

### CODE REVIEW

File: test/12_nullablearray.jl [hunk 1]
```diff
@@ -965,4 +965,66 @@ for isordered in (false, true)
     end
 end
 
+
+# Tests of vcat of NullableCategoricalArray
+
+for (CA,A) in ((:NullableCategoricalArray,:NullableArray),)
```
Comment (nalimilan) [inline test/12_nullablearray.jl]: I don't understand why you use a loop over only one set of values here.

**Comment by nalimilan** [comment]
> They are implemented together so at first I figured it would make sense to test them together. But then I realised that you've split all other tests of Nullable to its own file. The code is identical though.

Yeah, that's annoying, but in most cases there are slight variations which are very hard to handle without duplicating 90% of the code. Could make sense to split out the 100%-identical ones into a different file at some point.

File: test/12_nullablearray.jl (hunk 1 above)
Comment (gustafsson) [inline test/12_nullablearray.jl]: I only had one for-loop at first but wanted to hear your comments on the approach. The test code within the body of the for-loop is identical in the two files.

Comment (nalimilan) [inline test/12_nullablearray.jl]: OK. Then if you want just move these to a `13_arraycommon.jl` file (or something similar). We'll progressively move more common tests there.

Comment (nalimilan) [inline test/12_nullablearray.jl]: Oh, and BTW you don't need `@eval` here: `CA` and `A` can simply be types instead of symbols, and you don't need `$`.

Comment (gustafsson) [inline test/12_nullablearray.jl]: Oh right, I didn't think of skipping `@eval`, thanks!

File: test/13_arraycommon.jl [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+using CategoricalArrays: DefaultRefType
+using Compat
+
+typealias String Compat.ASCIIString
+
+# == currently throws an error for Nullables
+(==) = isequal
+
+
+# Tests of vcat of CategoricalArray amd NullableCategoricalArray
```
Comment (nalimilan) [inline test/13_arraycommon.jl]: and

File: test/13_arraycommon.jl [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+# Test that mergelevels handles mutually compatible ordering
+@test CategoricalArrays.mergelevels([6,3,4,7],[2,3,5,4],[2,4,8]) == ([6,2,3,5,4,7,8],true)
+@test CategoricalArrays.mergelevels([6,3,4,7],[2,3,6,5,4],[2,4,8]) == ([6,3,4,7,2,5,8],false)
+
+for (CA,A) in ((CategoricalArray,Array),(NullableCategoricalArray,NullableArray))
+    # Test that vcat of compact arrays use a reftype that doesn't overflow
```
Comment (nalimilan) [inline test/13_arraycommon.jl]: uses

File: test/13_arraycommon.jl [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+typealias String Compat.ASCIIString
+
+# == currently throws an error for Nullables
+(==) = isequal
+
+
+# Tests of vcat of CategoricalArray amd NullableCategoricalArray
+
+# Test that mergelevels handles mutually compatible ordering
+@test CategoricalArrays.mergelevels([6,3,4,7],[2,3,5,4],[2,4,8]) == ([6,2,3,5,4,7,8],true)
```
Comment (nalimilan) [inline test/13_arraycommon.jl]: In general, could you add spaces after commas? I'd rather keep a consistent style.

File: test/13_arraycommon.jl (hunk 2 above)
Comment (gustafsson) [inline test/13_arraycommon.jl]: ?

File: test/13_arraycommon.jl (hunk 3 above)
Comment (gustafsson) [inline test/13_arraycommon.jl]: ?

Comment (nalimilan) [inline test/13_arraycommon.jl]: Replace "use" with "uses".

File: test/13_arraycommon.jl (hunk 2 above)
Comment (nalimilan) [inline test/13_arraycommon.jl]: Replace "amd" with "and".

**Comment by nalimilan** [comment]
Thanks!

**Pull request merged**